debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SourceInfo.java

```diff
@@ -501,19 +505,42 @@ public void setOffset(Map<String, ?> sourceOffset) {
             this.restartEventsToSkip = longOffsetValue(sourceOffset, EVENTS_TO_SKIP_OFFSET_KEY);
             nextSnapshot = booleanOffsetValue(sourceOffset, SNAPSHOT_KEY);
             lastSnapshot = nextSnapshot;
-            this.databaseWhitelist = (String) sourceOffset.get(DATABASE_WHITELIST_KEY);
-            this.databaseBlacklist = (String) sourceOffset.get(DATABASE_BLACKLIST_KEY);
-            this.tableWhitelist = (String) sourceOffset.get(TABLE_WHITELIST_KEY);
-            this.tableBlacklist = (String) sourceOffset.get(TABLE_BLACKLIST_KEY);
+            if (sourceOffset.containsKey(DATABASE_INCLUDE_LIST_KEY)) {
+                this.databaseIncludeList = (String) sourceOffset.get(DATABASE_INCLUDE_LIST_KEY);
+            }
+            else {
+                this.databaseIncludeList = (String) sourceOffset.get(DATABASE_WHITELIST_KEY);
+            }
+            if (sourceOffset.containsKey(DATABASE_EXCLUDE_LIST_KEY)) {
+                this.databaseExcludeList = (String) sourceOffset.get(DATABASE_EXCLUDE_LIST_KEY);
+            }
+            else {
+                this.databaseExcludeList = (String) sourceOffset.get(DATABASE_BLACKLIST_KEY);
+            }
+            if (sourceOffset.containsKey(TABLE_INCLUDE_LIST_KEY)) {
+                this.tableIncludeList = (String) sourceOffset.get(TABLE_INCLUDE_LIST_KEY);
+            }
+            else {
+                this.tableIncludeList = (String) sourceOffset.get(TABLE_WHITELIST_KEY);
+            }
+            if (sourceOffset.containsKey(TABLE_EXCLUDE_LIST_KEY)) {
+                this.tableExcludeList = (String) sourceOffset.get(TABLE_EXCLUDE_LIST_KEY);
+            }
+            else {
+                this.tableExcludeList = (String) sourceOffset.get(TABLE_BLACKLIST_KEY);
+            }
         }
     }
 
     public static boolean offsetsHaveFilterInfo(Map<String, ?> sourceOffset) {
-        return sourceOffset != null &&
-                sourceOffset.containsKey(DATABASE_BLACKLIST_KEY) ||
+        return sourceOffset != null && (sourceOffset.containsKey(DATABASE_BLACKLIST_KEY) ||
```

Comment:
I added braces here for the ORs after the `&&` to fix the logic and avoid NPEs

Reply:
👍